src/npc.cpp

```diff
@@ -1083,19 +1107,35 @@ bool npc::wield(item* it, bool)
 
 bool npc::wield(item* it)
 {
-    if ( !weapon.is_null() ) {
+    const bool seen = g->u.sees( *this );
+    if( !weapon.is_null() ) {
         if ( volume_carried() + weapon.volume() <= volume_capacity() ) {
+            if( seen ) {
+                add_msg( m_info, _( "%1$s puts away the %2$s." ), name.c_str(), weapon.tname().c_str() );
```

Comment:
You could use `add_msg_if_npc` here. It has the advantage that it
- automatically checks whether the NPC is seen by the player (`seen`),
- replaces "\<npcname\>" in the message, so it's a simpler string: "\<npcname\> puts away the %s."